src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,187 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
+    public MyLinkedList(T value) {
+        addFirst(value);
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addFirst(value);
+            return;
+        }
+        tail.next = new Node<>(tail, value, null);
+        tail = tail.next;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkBound(index);
+        if (size == 0) {
+            addFirst(value);
+            return;
+        }
+        if (index == 0) {
+            head = new Node<>(null, value, head);
+            head.next.prev = head;
+            size++;
+            return;
+        }
+        if (size == index) {
+            addLast(value);
+            return;
+        }
+        Node<T> current = findByIndex(index);
+        Node<T> newNode = new Node<>(current.prev, value, current);
+        current.prev.next = newNode;
+        current.prev = newNode;
+        size++;
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkBound(index);
+        Node<T> current = findByIndex(index);
```

Comment:
check bound inside of a find by index method instead of calling them one by one every time